pyccel/parser/grammar/headers.tx

```diff
@@ -81,8 +81,15 @@ AvailableMacro:
 
 // TODO shall we add also results to the master call?
 
+//Extended Type with shape
+ExType: dtype=DataType ('*' prec=INT)? ('(' shape+=MacroValue[','] ')')? ( '(' 'order' '=' order = ID ')' )?;
+
+Shape: MacroValue | ':';
+
+MacroResult: name=ID ('(' shape+=Shape[','] ')')?;
```

Comment:
I would like to choose a more generic name for this if possible.

I suspect this object will be used more generally very soon. For example for the function `sgesvx` defined here in [pyccel](https://github.com/pyccel/pyccel/blob/master/pyccel/stdlib/internal/lapack.pyh#L27) and [here](https://www.kite.com/python/docs/scipy.linalg.lapack.sgesvx) in scipy. As you can see there are default arguments which are arrays (used to store memory inside the function call. I suspect that we will need to use `MacroResult` to recreate this. In this case it is no longer just a result, hence my request for a more generic name

Reply:
can you please suggest a name for this?

Reply:
`MacroShapedParameter` ? `ShapedID` ?

Reply:
Changed to `ShapedID` here bb2299d